packages/integrations/src/media-organizer/readarr/readarr-integration.ts

```diff
@@ -0,0 +1,114 @@
+import { logger } from "@homarr/log";
+import { z } from "@homarr/validation";
+
+import type { CalendarEvent } from "../../calendar-types";
+import { MediaOrganizerIntegration } from "../media-organizer-integration";
+
+export class ReadarrIntegration extends MediaOrganizerIntegration {
+  public async testConnectionAsync(): Promise<void> {
+    await super.handleTestConnectionResponseAsync({
+      queryFunctionAsync: async () => {
+        return await fetch(`${this.integration.url}/api`, {
+          headers: { "X-Api-Key": super.getSecretValue("apiKey") },
+        });
+      },
+    });
+  }
+
+  /**
+   * Gets the events in the Lidarr calendar between two dates.
+   * @param start The start date
+   * @param end The end date
+   * @param includeUnmonitored When true results will include unmonitored items of the Tadarr library.
+   */
+  async getCalendarEventsAsync(
+    start: Date,
+    end: Date,
+    includeUnmonitored = true,
+    includeAuthor = true,
+  ): Promise<CalendarEvent[]> {
+    const url = new URL(this.integration.url);
+    url.pathname = "/api/v1/calendar";
+    url.searchParams.append("start", start.toISOString());
+    url.searchParams.append("end", end.toISOString());
+    url.searchParams.append("unmonitored", includeUnmonitored ? "true" : "false");
```

Comment:
Maybe instead use `includeUnmonitored.toString()`?

Reply:
There is no such function

Reply:
She is not autocompleted but works, you can try it. It even gives you a comment when hovering over it